atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionOutcomeMetricsAssert.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl;
+
+import java.util.function.LongConsumer;
+
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.api.WritableAssertionInfo;
+import org.assertj.core.internal.Objects;
+
+import com.codahale.metrics.Metric;
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.logging.LoggingArgs;
+import com.palantir.tritium.metrics.registry.MetricName;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+
+public class TransactionOutcomeMetricsAssert extends
```

Comment:
For the record, I'm happy that this class exists, but I'm not certain that it belongs in `src/main/java` rather than `src/test/java`.

Reply:
I'd agree if this was in any other project. This is the `tests-shared` project though, which is used by other tests like the snapshot tx test (that can't see tests-shared's tests, as it shouldn't), so I think it's acceptable.